File: kotlin/darts/src/main/kotlin/Darts.kt
import kotlin.math.pow
import kotlin.math.sqrt

object Darts {

    fun score(x: Number, y: Number): Int {
        val distance = sqrt(x.toDouble().pow(2) + y.toDouble().pow(2))

        return when {
            distance <= 1.0 -> 10
            distance <= 5.0 -> 5
            distance <= 10.0 -> 1
            else -> 0
        }
    }
}

File: go/darts/darts.go
package darts

import "math"

func Score(x, y float64) int {
	distanceToCenter := distanceToCenter(x, y)
	switch {
	case distanceToCenter <= 1.0:
		return 10
	case distanceToCenter <= 5.0:
		return 5
	case distanceToCenter <= 10.0:
		return 1
	default:
		return 0
	}
}

func distanceToCenter(x, y float64) float64 {
	return math.Sqrt(math.Pow(math.Abs(x), 2) + math.Pow(math.Abs(y), 2))
}
